cosmo_tester/framework/util.py

```diff
@@ -489,16 +489,28 @@ def _raise_illegal(prop_path):
 
 @retrying.retry(stop_max_attempt_number=10, wait_fixed=5000)
 def assert_snapshot_created(manager, snapshot_id, attributes):
-    url = 'http://{ip}/api/{version}/snapshots/{id}'.format(
-        ip=manager.ip_address,
-        version=manager.api_version,
-        id=snapshot_id)
-    headers = {'tenant': attributes.cloudify_tenant}
-    auth = (attributes.cloudify_username, attributes.cloudify_password)
-    r = requests.get(url, auth=auth, headers=headers)
-    assert r.status_code == 200
-    snapshot = AttributesDict(r.json())
-    assert snapshot.status == 'created', 'Snapshot not in created status'
+    snapshots = manager.client.snapshots.list()
+
+    existing_snapshots = {
+        snapshot['id']: snapshot['status']
+        for snapshot in snapshots
+    }
+
+    assert snapshot_id in existing_snapshots.keys(), (
```

Comment:
no need for the `.keys()`, just `snapshot_id in existing_snapshots` is fine

Reply:
Yeah, I prefer to be explicit <smiles>